src/main/java/com/google/sps/servlets/StoreAccessory.java

```diff
@@ -0,0 +1,71 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.Accessory;
+import com.google.sps.data.Accessory.Type;
+import com.google.sps.data.AccessoryDatabase;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Stores Accessory data in database */
+@WebServlet("/store-accessory")
+public class StoreAccessory extends HttpServlet {
+  private static final String JSON_CONTENT_TYPE = "application/json";
+  private static final String SUBMIT_PARAMETER = "accessoryFormSubmit";
+  private static final String FILE_PARAMETER = "filepath";
+  private static final String TYPE_PARAMETER = "type";
+  private static final String HEIGHT_PARAMETER = "height";
+  private static final String WIDTH_PARAMETER = "width";
+  private static final String XPOS_PARAMETER = "xPos";
+  private static final String YPOS_PARAMETER = "yPos";
+  private static final String REDIRECTION_URL = "admin/StoreAccessory.html";
+  private static final String FILEPATH_ROOT = "images/accessories/";
+  private static Gson gson;
+  private File file;
+  private DatastoreService datastore;
+  private AccessoryDatabase accessoryDatabase;
+
+  @Override
+  public void init() {
+    this.gson = new Gson();
+    this.datastore = DatastoreServiceFactory.getDatastoreService();
+    this.accessoryDatabase = new AccessoryDatabase(datastore);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // just to ensure that submit was actually clicked
+    if (request.getParameter(SUBMIT_PARAMETER) != null) {
```

Comment:
this method is used a lot. Maybe move to a common place and refactor this to if(submitWasClicked(request)){ ... }

Reply:
+1, can we put a todo and issue for this?

Reply:
Done, added todo comment with issue link.